fontbe/src/post.rs

```diff
@@ -44,13 +45,43 @@ impl Work<Context, AnyWorkId, Error> for PostWork {
             .global_metrics
             .get()
             .at(static_metadata.default_location());
-        let postscript_names = &static_metadata.postscript_names;
-        let glyph_order = context.ir.glyph_order.get();
-        let mut post = Post::new_v2(
-            glyph_order
-                .names()
-                .map(|g| postscript_names.get(g).unwrap_or(g).as_str()),
-        );
+        // Whether to rename glyphs for 'production' using the provided remame map
+        let do_production_names = context.flags.contains(Flags::PRODUCTION_NAMES);
+        let rename_map = &static_metadata.postscript_names;
+        let mut seen = HashMap::new();
+        let final_glyph_names: Vec<_> = context
+            .ir
+            .glyph_order
+            .get()
+            .names()
+            .map(|g| {
+                if !do_production_names {
+                    // pass through the original name
+                    return g.to_string();
+                }
+                let mut name = rename_map.get(g).unwrap_or(g).to_string();
+                // Adobe Glyph List spec forbids any characters not in [A-Za-z0-9._]; it also says glyphs
+                // must not start with a digit or period (except .notdef) and shouldn't exceed 63 chars,
+                // but ufo2ft only enforces the first rule so we simply follow that.
+                // https://github.com/googlefonts/ufo2ft/blob/2f11b0f/Lib/ufo2ft/postProcessor.py#L220-L233
+                // https://github.com/adobe-type-tools/agl-specification
+                name.retain(|c| c.is_ascii_alphanumeric() || c == '.' || c == '_');
+                // make duplicates unique by adding a number suffix to match fonttools/ufo2ft:
+                // https://github.com/googlefonts/ufo2ft/blob/2f11b0f/Lib/ufo2ft/postProcessor.py#L239-L248
+                if let Some(n) = seen.get(&name) {
+                    let mut n = *n;
+                    while seen.contains_key(&format!("{}.{}", name, n)) {
```

Comment:
huh, why isn't it enough to just have `seen` be a map from `name` to `int`? then we would do,

```rust
let seen_n = seen.entry(&name).or_insert(0);
if *seen_n > 0 {
    name = format!("{name}.{seen_n}");
}
*seen_n += 1;
```